/*
 * Copyright(c) 2012-2018 Intel Corporation
 * SPDX-License-Identifier: BSD-3-Clause-Clear
 */
syntax = "proto3";
option cc_generic_services = true;
import "trace.proto";
package octf.proto.trace;

message ParsedEvent {
    message IO {
        /** Address of IO in sectors */
        uint64 lba = 1;

        /** Size of IO in sectors */
        uint32 len = 2;

        /** IO class of IO */
        uint32 ioClass = 3;

        /** Device ID */
        uint64 deviceId = 4;

        /** Operation type: read, write, trim */
        IoType operation = 5;

        /** Flush flag */
        bool flush = 6;

        /** FUA flag */
        bool fua = 7;

        /** Flags indicates if IO ended with an error */
        bool error = 10000;

        /** IO Latency */
        uint64 latency = 10001;

        /** IO queue depth */
        uint64 qd = 10002;
    }

    message DeviceInfo {
        /** Device Name */
        string name = 2;
    }

    message FileInfo {
        /** File ID */
        uint64 id = 2;

        /** File offset in sectors */
        uint64 offset = 4;

        /** File size in sectors */
        uint64 size = 5;
    }

    /** Header Information */
    EventHeader header = 1;

    /** IO Basic Information */
    IO io = 2;

    /** Device Information */
    DeviceInfo device = 3;

    /** File Information */
    FileInfo file = 4;
}
